repository/src/main/java/tech/ydb/yoj/repository/db/cache/TransactionLocal.java

```diff
@@ -33,14 +35,39 @@ public <X> X instance(@NonNull Supplier<X> supplier) {
         return (X) singletons.computeIfAbsent(supplier, Supplier::get);
     }
 
+    /**
+     * <strong>Warning:</strong> Unlike {@link #log()}, this method is not intended to be used by end-users, only by the YOJ implementation itself.
+     * <p>Also, projection support <em>might</em> be dropped or seriously reworked in YOJ 3.x, so please do not rely on this implementation detail.
+     */
     public ProjectionCache projectionCache() {
         return instance(projectionCacheSupplier);
     }
 
+    /**
+     * @deprecated Unsupported and will be removed in YOJ 3.0.0. Use the {@link #firstLevelCache(TableDescriptor)} instead.
+     *
+     * @throws UnsupportedOperationException always
+     */
+    @Deprecated(forRemoval = true)
+    @SuppressWarnings("rawtypes")
     public FirstLevelCache firstLevelCache() {
-        return instance(firstLevelCacheSupplier);
+        throw new UnsupportedOperationException("TransactionLocal.firstLevelCache() is not supported");
```

Comment:
Don't throw an exception here, because when updating the library, colleagues might get exceptions in production since this code is being used. I suggest simply removing this method.

Reply:
So we'll trade a runtime exception for a compilation error? I'm OK if Cloud Java is OK with this ;-)

Reply:
I prefer a compile-time error over a runtime error😄